tidy3d/components/eme/simulation.py

```diff
@@ -542,9 +553,34 @@ def _post_init_validators(self) -> None:
         self._validate_modes_size()
         self._validate_sweep_spec()
         self._validate_symmetry()
+        self._validate_constraint()
         # self._warn_monitor_interval()
         log.end_capture(self)
 
+    def _validate_constraint(self):
+        """Constraint can be slow with too many modes. Warn in this case."""
+        constraint = self.constraint
+        max_num_modes = self.max_num_modes
+        if constraint is not None and max_num_modes > WARN_CONSTRAINT_NUM_MODES:
+            log.warning(
+                f"The simulation has 'constraint='{constraint}'', and the maximum "
```

Comment:
I think the additional single quotes `'` around `{constraint}` will mean this will not render properly. You probably need to escape the single quotes. `\'` but double check this.